File: The Braithwaite House Escape/sol.js
function max(a, b) {
    return a >= b ? a : b;
}

function solve() {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    rl.question('', input => {
        const [n, m, r, c] = input.split(' ').map(Number);
        console.log(max(n - r, r - 1) + max(m - c, c - 1));
        rl.close();
    });
}

solve();

File: The Braithwaite House Escape/sol.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func max(a, b int) int {
	if a >= b {
		return a
	}
	return b
}

func solve() {
	reader := bufio.NewReader(os.Stdin)
	var n, m, r, c int

	fmt.Fscan(reader, &n, &m, &r, &c)
	fmt.Println(max(n-r, r-1) + max(m-c, c-1))
}

func main() {
	solve()
}
